### PR TITLE
Add markdown-it-py recipe

### PR DESCRIPTION
**Pull request by chrisjsewell**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by chrisjsewell** [comment]
@conda-forge/help-python this is ready for review thanks

File: recipes/markdown-it-py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "925da55cba665cb9808ef851885993f856c678c8009077a59ab3134d7dac876f"
+
+build:
+  number: 0
+  entry_points:
+    - markdown-it = markdown_it.cli.parse:main
+    - markdown-it-bench = markdown_it.cli.benchmark:main
+  noarch: "python"
```
Comment (npavlovikj) [inline recipes/markdown-it-py/meta.yaml]: You can just use `noarch: python` here in future.

File: recipes/markdown-it-py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (npavlovikj) [inline recipes/markdown-it-py/meta.yaml]: For future reference, only `script: "{{ PYTHON }} -m pip install . -vv"` is enough here..

**Comment by npavlovikj** [comment]
Thank you for contributing @chrisjsewell 🎉

**Comment by chrisjsewell** [comment]
Thanks 😊

**Pull request merged**